### ISSUE TITLE
Support for dataclass_transform converters

### ISSUE THREAD

**Issue by AdrianSosic** [astral-sh/ty#972]
### Summary

The following code throws a false-positive:

```python
from attrs import define, field


@define
class C:
    x: float = field(converter=float)


C("1")
```

```bash
error[invalid-argument-type]: Argument is incorrect
 --> baybe/ty.py:9:3
  |
9 | C("1")
  |   ^^^ Expected `int | float`, found `Literal["1"]`
  |
info: rule `invalid-argument-type` is enabled by default
```

The issue is that attrs converters don't seem to be supported yet, which are handled via the attrs plugin in mypy. I'm aware that there is already a PR to enable dataclass support (#111) but since converters are not available via dataclasses, I think this is probably a separate issue.

**Comment by Darkdragon84** [comment]
there must be something in the typing standard about that, perhaps via [PEP681](https://peps.python.org/pep-0681/), because pyright supports attrs converters, and they only ever support features that are in the standard.

**Comment by erictraut** [comment]
Yes, support for `converter` was added to the typing spec. See [here](https://discuss.python.org/t/discuss-adding-converter-to-dataclass-transform/55125) for the discussion.

**Comment by Darkdragon84** [comment]
> Yes, support for `converter` was added to the typing spec. See [here](https://discuss.python.org/t/discuss-adding-converter-to-dataclass-transform/55125) for the discussion.

Ah I didn't know PEPs could be amended. TIL 👍

**Comment by sharkdp** [comment]
The new section in the spec for this is here: https://typing.python.org/en/latest/spec/dataclasses.html#converters
